### PR TITLE
emacs: bump

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

**Comment by extrowerk** [comment]
fixes https://github.com/haikuports/haikuports/issues/2743

File: app-editors/emacs/emacs-26.1.recipe [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+{
+	# Hard links not supported on BFS.
+	sed -i 's/ln -f/ln -s/g' src/Makefile.in
+}
+
+BUILD()
+{
+	export LDFLAGS="-lnetwork -lbsd"
+	CFLAGS="-D_BSD_SOURCE"
+	export HOME=/boot/home
```
Comment (alaviss) [inline app-editors/emacs/emacs-26.1.recipe]: This should be contained inside the build folder. We don't want it to mess with user's home when they build the recipe without chroot.

Also:

```diff
-	export LDFLAGS="-lnetwork -lbsd"
-	CFLAGS="-D_BSD_SOURCE"
+	export LIBS="-Wl,--as-needed -lnetwork -lbsd"
+	export CPPFLAGS="-D_BSD_SOURCE"
```

`-Wl,--as-needed` will avoid linking those libraries if they are unneeded.

File: app-editors/emacs/emacs-26.1.recipe [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		rm -rf src/$f
+	done
+
+	make install
+
+	# ctags fix
+	mv $binDir/ctags $binDir/ctags.emacs
+
+	# Cleanup
+	find $dataDir -type f -name "*.el.gz" -delete
```
Comment (alaviss) [inline app-editors/emacs/emacs-26.1.recipe]: What are those files?

Comment (extrowerk) [reply app-editors/emacs/emacs-26.1.recipe]: The emacs source code. Emacs provides his source as gzip files. Some packagers creates an own subpackage for them.
They aren't needed for normal use.

**Pull request merged**